### PR TITLE
fix: Prevent IllegalArgumentException when sortable columns are remov…

### PR DESCRIPTION
**Pull request by TatuLund**
…ed and recreated

Fixes: https://github.com/vaadin/vaadin-grid/issues/1969

### CODE REVIEW

**Comment by tomivirkki** [comment]
Missing tests

**Comment by tomivirkki** [comment]
The issue is likely caused by https://github.com/vaadin/vaadin-grid/issues/1691 which means that the `<vaadin-grid> 's` `_sorters` array keeps references to sorters of removed column instances. If that's the case, we should fix the root cause of the problem (in the Web Component)

**Comment by tomivirkki** [comment]
Well...this change might actually still be required to fully fix the issue. If a sorted column is removed from a detached Flow Grid, there's no active Web Component to notify of the change so you might end up with a wrongly synchronized server-side state if the sortOrder isn't updated here.

File: vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java [hunk 1]
```diff
@@ -2929,6 +2929,8 @@ public void removeColumn(Column<T> column) {
         Objects.requireNonNull(column, "column should not be null");
 
         ensureOwner(column);
+        List<GridSortOrder<T>> order = new ArrayList<>();
+        setSortOrder(order, false);
```
Comment (tomivirkki) [inline vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java:2933]: Wouldn't this reset all sorting from the Grid whenever a single column gets removed?

Comment (TatuLund) [reply vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java:2933]: Yes, and I think it is the correct behavior.

**Comment by hindog** [comment]
FYI, I'm using Vaadin 14.4.7 (NPM) and currently hitting this issue.  Is there a work-around for this?  

One attempt I had tried was to try 'hiding' instead of 'removing' the columns with `column.setVisible(false)` followed by`grid.sort(null)` to clear the sort.  Let's say I have column "A" and column "B":

- Sort column "A"
- "Remove" column "A" by doing: `columnA.setVisible(false); grid.clear(null)`
- Sort column "B"
- My data provider will receive sorts for "A" and "B", when I would only expect it for "B" because I did `grid.clear(null)` and hid the column.

**Review by tulioag — APPROVED** [review]

**Pull request merged**